### DISCUSSION TITLE
pzem-004t serial connection

### DISCUSSION THREAD

**Discussion by freedomwarrior** [micropython/micropython#14453] [ESP8266] [answered]
Hello.
Recently got **pzem-004t** device and now I'm trying to connect to this device via serial and get some measurements.
But I faced the problem that I can't send commands and read values from the device and at the same time be in REPL to see some information(errors, values, etc).
I'm using [this](https://github.com/jacopoRodeschini/PZEM-004T) library and author said:
`In the ESP8266 there is only one available UART port, thus if you use the RELP terminal is necessary to detach it for reading values from the PZEM-004T and then attach it to communicate with the terminal `
So, is there some way to bypass this restriction ?

**Comment by robert-hh** [comment]
You can use webrepl for the REPL interface and detach REPL from UART with dupterm.
Webrepl: https://github.com/micropython/webrepl

**Comment by freedomwarrior** [comment]
> You can use webrepl for the REPL interface and detach REPL from UART with dupterm. Webrepl: https://github.com/micropython/webrepl

Working, thank you for workaround.
But I faced with another problem, when I try to read data from pzem-004t, I got only half of data, only first 13 bytes from 25 that must be.

**Reply by robert-hh** [reply, marked as answer]
The buffer at the serial port has a default length of 15 bytes. You can increase that using the rxbuf option of the UART constructor of uart.init.

**Comment by freedomwarrior** [comment]
Thanks again, working like a charm.
